libs/langchain/langchain/vectorstores/elastic_vector_search.py

```diff
@@ -156,12 +156,22 @@ def __init__(
         self.index_name = index_name
         _ssl_verify = ssl_verify or {}
         try:
-            self.client = elasticsearch.Elasticsearch(elasticsearch_url, **_ssl_verify)
+            self.client = elasticsearch.Elasticsearch(
+                elasticsearch_url,
+                **_ssl_verify,
+                headers={"user-agent": self.get_user_agent()},
+            )
         except ValueError as e:
             raise ValueError(
                 f"Your elasticsearch client string is mis-formatted. Got error: {e} "
             )
 
+    @staticmethod
+    def get_user_agent() -> str:
+        from langchain import __version__
```

Comment:
in case you care __version__ evaluates to empty string when not installed but developing w/ code base